backend/danswer/background/celery/apps/app_base.py

```diff
@@ -146,6 +146,8 @@ def on_celeryd_init(sender: Any = None, conf: Any = None, **kwargs: Any) -> None
 def wait_for_redis(sender: Any, **kwargs: Any) -> None:
     """Waits for redis to become ready subject to a hardcoded timeout.
     Will raise WorkerShutdown to kill the celery worker if the timeout is reached."""
+    if MULTI_TENANT:
```

Comment:
Understand we don't need the checks for multi-tenant/cloud, but conditionals like this should be as high in the call stack as possible, not embedded in tool functions.

Reply:
Completely agree! Moved all these checks up to the worker init